frontend/src/metabase/search/types.ts

```diff
@@ -55,7 +57,7 @@ type SidebarFilterType = "dropdown" | "toggle";
 
 interface SearchFilter<T extends FilterTypeKeys = any> {
   type: SidebarFilterType;
-  label: string;
+  label: () => string;
```

Comment:
Adjusting the typing here to account for the dynamic label that we use in the SQL label. I personally think it would be better to have a single type than have a `string | () => string` and then have to check types in.